prov/gni/src/gnix_mbox_allocator.c

```diff
@@ -51,74 +53,45 @@
  * @return -FI_EINVAL	if an invalid parameter was given
  * @return -FI_EIO	if an error occurred while opening the /proc/mounts
  * file.
- * @return -FI_ENOMEM	if an error occurred while allocating space for the text
- * from the file.
  */
 static int __find_huge_page(size_t page_size, char **directory)
 {
-	int bytes_read = -FI_EINVAL;
+	int rc = -FI_EINVAL;
 	struct statfs pg_size;
-	size_t size = 1024;
-	char error_buf[256];
-	char *line, *field;
-	char *saveptr;
-	char *error;
+	struct mntent *mntent;
 	FILE *fd;
 
 	if (!directory || !page_size) {
 		GNIX_WARN(FI_LOG_EP_CTRL,
 			  "Invalid page_size or directory provided.\n");
-		bytes_read = -FI_EINVAL;
-		goto err_fopen;
+		return -FI_EINVAL;
 	}
 
-	*directory = NULL;
-
-	fd = fopen("/proc/mounts", "r");
-	if (!fd) {
-		error = strerror_r(errno, error_buf, sizeof(error_buf));
-		GNIX_WARN(FI_LOG_EP_CTRL, "Error opening /proc/mounts: %s\n",
-			  error);
-		bytes_read = -FI_EIO;
-		goto err_fopen;
+	fd = setmntent ("/proc/mounts", "r");
+	if (fd == NULL) {
+		GNIX_WARN(FI_LOG_EP_CTRL,
+			  "Invalid page_size or directory provided.\n");
```

Comment:
This error message doesn't seem quite right.  It's probably just fine to use the original message.